### PR TITLE
ext_authz: fix body buffering feature on ExtAuthZ per-route filter

### PR DESCRIPTION
**Pull request by agrawroh**
## Background

The [with_request_body](https://arc.net/l/quote/sxyvvazy) feature on per-route ExtAuthZ filter is not working as expected.

**See:** https://github.com/envoyproxy/envoy/issues/31436

## Changes

This PR fixes the body propagation when [with_request_body](https://arc.net/l/quote/cdhvwyic) is enabled on a per-route ExtAuthZ filter. Current implementation reads the flags for partial body and max bytes size only from the main HTTP filter and discards these configs on the per-route ExtAuthZ filter. 

Commit Message: fix body buffering feature on ExtAuthZ per-route filter
Additional Description:add new private fields to hold the final state for the max bytes and whether partial body is allowed in the ExtAuthZ filter
Risk Level: Low
Testing: Added Tests
Docs Changes: N/A
Release Notes: Added
Platform Specific Features: N/A
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: test/extensions/filters/http/ext_authz/ext_authz_test.cc [hunk 1]
```diff
@@ -3004,7 +3004,7 @@ TEST_F(HttpFilterTest, PerRouteCheckSettingsWorks) {
   failure_mode_allow: false
   )EOF");
 
-  // Set allow_partial_message to true and max_request_bytes to 10 on the per-route filter.
+  // Set allow_partial_message to true and max_request_bytes to 5 on the per-route filter.
```
Comment (tyxia) [inline test/extensions/filters/http/ext_authz/ext_authz_test.cc:3007]: Not your change, Could you please also fix the typo in line#2977 while you are already here?

**Review by tyxia — APPROVED** [review]
LGTM, Thank you for fixing this

File: test/extensions/filters/http/ext_authz/ext_authz_test.cc (hunk 1 above)
Comment (agrawroh) [reply test/extensions/filters/http/ext_authz/ext_authz_test.cc:3007]: Done! Thanks for pointing it out.

**Review by tyxia — APPROVED** [review]
LGTM, Thanks!

Defer to @ggreenway  for review and merge

**Review by adisuissa — COMMENTED** [review]
Drive-by comment, please add a release note.

**Review by ggreenway — CHANGES_REQUESTED** [review]
LGTM. Pleas add a release note.

/wait

**Comment by agrawroh** [comment]
> The [with_request_body](https://arc.net/l/quote/sxyvvazy) feature on per-route ExtAuthZ filter is not working as expected.

Added, Thanks!

**Review by ggreenway — APPROVED** [review]

**Pull request merged**